addons/account/models/account_move.py

```diff
@@ -4265,37 +4244,25 @@ def name_get(self):
-        if tracked_field.get('field_type') in ('date', 'datetime'):
-            return {
-                'old_value': format_date(self.env, fields.Datetime.from_string(tracked_field.get('old_value_datetime'))),
-                'new_value': format_date(self.env, fields.Datetime.from_string(tracked_field.get('new_value_datetime'))),
-            }
-        elif tracked_field.get('field_type') in ('one2many', 'many2many', 'many2one'):
-            return {
-                'old_value': tracked_field.get('old_value_char', ''),
-                'new_value': tracked_field.get('new_value_char', '')
-            }
-        else:
-            return {
-                'old_value': [val for key, val in tracked_field.items() if 'old_value' in key][0], # Get the first element because we create a list like ['Elem']
-                'new_value': [val for key, val in tracked_field.items() if 'new_value' in key][0], # Get the first element because we create a list like ['Elem']
-            }
-
-    def _get_tracking_field_string(self, fields):
-        ARROW_RIGHT = '<span aria-label="Changed" class="fa fa-long-arrow-alt-right" role="img" title="Changed"></span>'
-        msg = '<ul>'
-        for field in fields:
-            redirect_link = '<a href=# data-oe-model=account.move.line data-oe-id=%d>#%d</a>' % (field['line_id'], field['line_id']) # Account move line link
-            if field.get('error', False):
-                msg += '<li>%s: %s</li>' % (
-                    field['field_error'],
-                    _('A modification has been operated on the line %s.', redirect_link)
-                )
-            else:
-                msg += '<li>%s: %s %s %s (%s)</li>' % (field['field_name'], field['old_value'], ARROW_RIGHT, field['new_value'], redirect_link)
-        msg += '</ul>'
-        return msg
+    def _mail_track(self, tracked_fields, initial):
+        changes, tracking_value_ids = super()._mail_track(tracked_fields, initial)
+        if len(changes) > len(tracking_value_ids):
+            for i, changed_field in enumerate(changes):
+                if tracked_fields[changed_field]['type'] in ['one2many', 'many2many']:
+                    field = self.env['ir.model.fields']._get(self._name, changed_field)
+                    tracking_value_ids.insert(i, [
+                        0,
+                        0,
+                        {
+                            'field': field.id,
+                            'field_desc': tracked_fields[changed_field]['string'],
+                            'field_type': tracked_fields[changed_field]['type'],
+                            'tracking_sequence': 100 if isinstance(getattr(self._fields[changed_field], 'tracking', False), bool) else getattr(self._fields[changed_field], 'tracking', 100),
+                            'old_value_char': len(initial[changed_field]),
+                            'new_value_char': _('%(rec_count)s records', rec_count=len(self[changed_field]))
```

Comment:
Why only the number of records?

```suggestion
                            'old_value_char': ', '.join(initial[changed_field].mapped('name')),
                            'new_value_char': ', '.join(self[changed_field].mapped('name')),
```

Reply:
Also, it doesn't seem to track the values everywhere. For instance, changing the tax on an invoice line will trigger the change of tags on that line, but it won't be logged for the related tax line.

Reply:
Are you sure? Did you use a chart with tags defined? and remember, the logging is only for modified entries, so you have to reset to draft.

Reply:
> Did you use a chart with tags defined?

Yes

> the logging is only for modified entries, so you have to reset to draft.

I posted an invoice with one line, reset to draft, and changed the tax on the invoice line.
The tax tags on both the tax line and the base line changed, but only the change on the base line was logged, not on the tax line.

Reply:
Ah I see what you mean, it's weird. If you change the tags directly on the tax line, the message is logged. 
I suspect it's because its actually a new line, I will explore further...
But, do we really want an additional message?

Reply:
I'm pretty sure we want the information to be logged one way or another

Reply:
Indeed, the tax lines are actually new lines. They are created by an `api.onchange` method in `account.move` - hence there are no "old" values to compare with, when `write` on `account.move.line` is called.

Try the following:
1. Edit the tax on an invoice line
2. Edit the tags on the tax line journal item - it will log a change and the id of the tax line
3. Edit the tax on the same invoice line from 1.
4. Edit the tags on the "same" tax line (which will have changed by now) & note the id of the tax line

Since direct changes to the tax line are logged - IMO this is adequate information for the user.